gradio/routes.py

```diff
@@ -303,7 +303,7 @@ def login(form_data: OAuth2PasswordRequestForm = Depends()):
 
         @app.head("/", response_class=HTMLResponse)
         @app.get("/", response_class=HTMLResponse)
-        def main(request: fastapi.Request, user: str = Depends(get_current_user)):
+        async def main(request: fastapi.Request, user: str = Depends(get_current_user)):
```

Comment:
Should be good to change to async since TemplateResponse doesn't block

Reply:
Thanks for confirming